kratos/utilities/read_materials_utility.cpp

```diff
@@ -245,32 +286,177 @@ void ReadMaterialsUtility::AssignPropertyBlock(Parameters Data)
             const auto& r_input_var = KratosComponents<Variable<double>>().Get(input_var_name);
             const auto& r_output_var = KratosComponents<Variable<double>>().Get(output_var_name);
 
-            CheckIfOverwritingTable(*p_prop, r_input_var, r_output_var);
+            if (Data.Has("model_part_name")) {
+                CheckIfOverwritingTable(*p_prop, r_input_var, r_output_var);
+            }
 
             for (IndexType i = 0; i < table_param["data"].size(); ++i) {
                 table.insert(table_param["data"][i][0].GetDouble(),
                             table_param["data"][i][1].GetDouble());
             }
-            p_prop->SetTable(r_input_var, r_output_var, table);
+            pNewProperty->SetTable(r_input_var, r_output_var, table);
+        }
+    } else {
+        KRATOS_INFO("Read materials") << "No tables defined for material ID: " << pNewProperty->Id() << std::endl;
+    }
+
+    KRATOS_CATCH("");
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void ReadMaterialsUtility::CreateSubProperties(
+    ModelPart& rModelPart,
+    Parameters Data,
+    Properties::Pointer& pNewProperty
+    )
+{
+    KRATOS_TRY;
+
+    if (Data.Has("sub_properties")) {
+
+        auto& r_list_sub_properties = pNewProperty->GetSubProperties();
+
+        const std::size_t number_of_subproperties = Data["sub_properties"].size();
+
+        // We assign the subproperties now
+        for(std::size_t i_sub_prop=0; i_sub_prop < number_of_subproperties; ++i_sub_prop) {
+            // Copy of the current parameters
+            Parameters sub_prop = Data["sub_properties"][i_sub_prop];
+
+            // We get the adress if any
+            const std::string& r_use_existing_property = sub_prop.Has("use_existing_property") ? sub_prop["use_existing_property"].GetString() : "";
```

Comment:
this could be rewritten more clear. 

as i understand there are two options:
1 - use existing --> in this case either it exists or it is an error
2 - create new, than you use the function CreateProperty to read a new one.

then i would realyly do

~~~py
     if(sub_prop.Has("use_existing_property"))
     {
          ... find it and assign it ...
     }
     else
     {
          call function CreateProperty(   passing the flag "treat_as_subproperty" )
     }
~~~

Reply:
This is what is done